specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+        
+        /// Indicates that the context menu was created for a video element.
+        COREWEBVIEW2_CONTEXT_KIND_VIDEO,
+    } COREWEBVIEW2_CONTEXT_KIND;
+
+    /// Specifies the menu item kind
+    /// for the `ICoreWebView2StagingContextMenuItem::get_Kind` method
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND
+    {
+        /// Specifies a normal menu item kind.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL,
+
+        /// Specifies a checkbox menu item kind. `ContextMenuItem` objects of this kind
+        /// will need the `IsChecked` property to determine current state of the checkbox.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECKBOX,
+
+        /// Specifies a radio button menu item kind. `ContextMenuItem` objects of this kind
+        /// will need the `IsChecked` property to determine current state of the radio button.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO,
+
+        /// Specifies a separator menu item kind. `ContextMenuItem` objects of this kind
+        /// are used to signal a visual separator with no functionality.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SEPARATOR,
+
+        /// Specifies a submenu menu item kind. `ContextMenuItem` objects of this kind will contain
+        /// a `ContextMenuItemCollection` of its children `ContextMenuItem` objects.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SUBMENU,
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND;
+
+    /// Represents a context menu item of a context menu displayed by WebView.
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown
+    {
+        /// Gets the localized label for the `ContextMenuItem`.
+        [propget] HRESULT Label([out, retval] LPWSTR* value);
+
+        /// Gets the unlocalized name for the `ContextMenuItem`. Use this to distinguish items with 
+        /// same Descriptor such as Extension items and items with Other as descriptor.
+        [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+        /// Gets the descriptor for the `ContextMenuItem`.
+        [propget] HRESULT Descriptor([out, retval] COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR* value);
+
+        /// Gets the Command ID for the `ContextMenuItem`. Use this to report the `SelectedCommandId` in `ContextMenuRequested` event.
+        [propget] HRESULT CommandId([out, retval] INT32* value);
+
+        /// Gets the keyboard shortcut for this ContextMenuItem. It will be the empty
+        /// string if there is no keyboard shortcut.
+        /// This is text intended to be displayed to the end user to show the keyboard shortcut. 
+        /// For example this property is `Ctrl+Shift+I` for the "Inspect" `ContextMenuItem`.
+        [propget] HRESULT ShortcutKeyDescription([out, retval] LPWSTR* value);
+
+        /// Gets the Icon for the `ContextMenuItem` in PNG format in the form of an IStream.
+        [propget] HRESULT Icon([out, retval] IStream** value);
```

Comment:
Will the stream be rewound to the start of the PNG? Or does the app have to seek the stream back to 0 before decoding?

Reply:
The stream will be rewound to the start of the PNG by CoreWebView2.

Reply:
Please note that in the documentation.